File: Source/prj/reg/PackageInfo.py

from mtm.util.Assert import *

class PackageFolderInfo:
    def __init__(self):
        self.path = None
        self.packages = []

class PackageInfo:
    def __init__(self):
        self.name = None
        # Might be null
        self.installInfo = None

class PackageInstallInfo:
    def __init__(self):
        self.installDate = None
        self.releaseInfo = None

File: UnityPlugin/Projeny/Main/Models/PackageInfo.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.IO;
using System.Linq;
using System.Text;
using UnityEditor;
using UnityEditor.Callbacks;
using UnityEditor.SceneManagement;
using UnityEditorInternal;
using UnityEngine;
using Projeny.Internal;

namespace Projeny
{
    [Serializable]
    public class PackageFolderInfo
    {
        public string Path;
        public List<PackageInfo> Packages = new List<PackageInfo>();
    }

    [Serializable]
    public class PackageInfo
    {
        public string Name;
        public PackageInstallInfo InstallInfo;
        public string FullPath;
    }

    [Serializable]
    public class PackageInstallInfo
    {
        public string InstallDate;
        public long InstallDateTicks;

        public ReleaseInfo ReleaseInfo;
    }
}

